pandas/core/frame.py

```diff
@@ -5058,7 +5065,12 @@ def _sanitize_column(self, value) -> tuple[ArrayLike, BlockValuesRefs | None]:
 
         if is_list_like(value):
             com.require_length_match(value, self.index)
-        return sanitize_array(value, self.index, copy=True, allow_2d=True), None
+
+        if not pa_version_under10p1 and isinstance(value, pa.lib._PandasConvertible):
```

Comment:
Was this design taken from a prior art? I would advise against reaching into the internals of third party libraries like pyarrow

@mroeschke may know best here

Reply:
In practice, this is the base class for Array/ChunkedArray/RecordBatch/Table. But I think in this case (values for a single column) we only would want to allow setting pyarrow Array and ChunkedArray? 
(in which case I think you can do an `isinstance` check with just those classes)

Reply:
I completely agree. I replaced the test with an `isinstance` check

Reply:
I have a strange bug (an overflow maybe) in `test_find_result_type_uint_int `only on Pyarrow Nightly job and another PRs have exactly the same, as for example:
- https://github.com/pandas-dev/pandas/pull/59071
- https://github.com/pandas-dev/pandas/pull/59065

Is it a common problem on the CI ?
Thanks in advance,

```
_________________ test_find_result_type_uint_int[right5-int16] _________________
[gw1] linux -- Python 3.11.9 /home/runner/micromamba/envs/test/bin/python3.11

right = np.int16(300), result = <class 'numpy.int16'>

    @pytest.mark.parametrize(
        "right,result",
        [
            (0, np.uint8),
            (-1, np.int16),
            (300, np.uint16),
            # For floats, we just upcast directly to float64 instead of trying to
            # find a smaller floating dtype
            (300.0, np.uint16),  # for integer floats, we convert them to ints
            (300.1, np.float64),
            (np.int16(300), np.int16 if np_version_gt2 else np.uint16),
        ],
    )
    def test_find_result_type_uint_int(right, result):
        left_dtype = np.dtype("uint8")
>       assert find_result_type(left_dtype, right) == result
E       AssertionError
```

Reply:
yes, @droussea2001 at least this shouldn't be just because of the changes you made here, maybe you won't know until the CI of other PRs passes.

Reply:
Ok thanks !